From: Gerald Schaefer <geraldsc@de.ibm.com>
Subject: IBM Codestream linux-2.6.16 october2005, patch 02-03

  Description: z90crypt: analysis revealed a possible memory overlay
  Symptom:     kernel oops
  Problem:     incorrect size for an array
  Solution:    use sizeof() as the array dimension
  Problem-ID:  22773
  Archs:       s390-31, s390-64

Acked-by: Hannes Reinecke <hare@suse.de>

--- linux-2.5/drivers/s390/crypto/z90main.c	30 Mar 2006 22:01:11 -0000	1.87.2.2
+++ linux-2.5/drivers/s390/crypto/z90main.c	30 Mar 2006 22:05:26 -0000	1.87.2.3
@@ -3283,7 +3283,7 @@
 {
 	int rv, dv, i, index, length;
 	unsigned char psmid[8];
-	static unsigned char loc_testmsg[548];
+	static unsigned char loc_testmsg[sizeof(MCL3_testmsg)];
 	struct CPRBX *cprbx_p;
 
 	index = devPtr->dev_self_x;

